flink-connectors/flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSGeneralUtil.java

```diff
@@ -374,4 +371,34 @@ public static void closeResources(SdkAutoCloseable... resources) {
             throw exception;
         }
     }
+
+    public static void validateAwsCredentials(Properties config) {
+        try {
+            validateAwsConfiguration(config);
+            getCredentialsProvider(config).resolveCredentials();
+        } catch (IllegalStateException | IllegalArgumentException | SdkClientException e) {
+            throw new AWSAuthenticationException(
+                    String.format(
+                            "Failed to create client using %s provider",
+                            getCredentialProviderType(
+                                    config, AWSConfigConstants.AWS_CREDENTIALS_PROVIDER)),
+                    e);
+        }
+    }
```

Comment:
Again, how about if there are other issues thrown here that are not related to authentication? `SdkClientException`

>Direct Known Subclasses:
AbortedException, ApiCallAttemptTimeoutException, ApiCallTimeoutException, Crc32MismatchException, EndpointDiscoveryFailedException, ExpiredTokenException, NonRetryableException, RetryableException, SdkJsonGenerator.JsonGenerationException


Consider letting these bubble up, or wrapping more specific exceptions